### PR TITLE
Fix entities getting launch into the air

### LINKED ISSUE #2707: Knockback is so much
### Issue description
Knockback is so much according to vanilla mcpe

### Steps to reproduce the issue
1. Place a entity (ex: villager)
2. Get a enchanted item with knockback
3. Attack to entity
4. Compare this with vanilla mcpe

### OS and versions

* PocketMine-MP: 3.5.x
* PHP: 7.2.0
* Server OS: Windows 10
* Game version: Win10 1.8

### Crashdump, backtrace or other files

[in pmmp](https://youtu.be/h4PciGEe-Ck)
[in vanilla](https://youtu.be/_oT3zWWgFk0)

### PR DESCRIPTION
**Pull request by JavierLeon9966**
## Introduction

This PR fix entities getting launch into the air when getting knockback.
### Relevant issues

Fixes #2707
### API changes

None
### Behavioural changes

None
## Backwards compatibility

Added a verticalLimit only accepts float and default to null
## Follow-up

None
## Tests

Script: https://wandbox.fetus.jp/permlink/X0bQydRSGzmmxorm

### CODE REVIEW

**Comment by dktapps** [comment]
This will break knockback for everyone using it for non-standard purposes (e.g. launchpads).

**Comment by JavierLeon9966** [comment]
Oh true

**Comment by JavierLeon9966** [comment]
Well then what to do about this PR?

**Comment by JavierLeon9966** [comment]
Also this is for PM4 So it’ll not BC in PM3. According to Twisted nothing is considered BC in PM4

**Comment by JavierLeon9966** [comment]
I have a idea to conserve BC

**Comment by dktapps** [comment]
Your "tests" section is not showing any tests, only telling what I can already see in the diff with my own eyes.

File: src/entity/Living.php [hunk 1]
```diff
@@ -503,7 +503,7 @@ protected function doHitAnimation() : void{
 		$this->broadcastAnimation(new HurtAnimation($this));
 	}
 
-	public function knockBack(float $x, float $z, float $base = 0.4) : void{
+	public function knockBack(float $x, float $z, float $base = 0.4, bool $vanilla = false) : void{
```
Comment (dktapps) [inline src/entity/Living.php:506]: This is unnecessarily limiting. A vertical limit parameter would be more suitable.

Comment (JavierLeon9966) [reply src/entity/Living.php]: Now is changed, and don’t mind my typo, stupid autocorrect

**Pull request closed without merging**